src/index.ts

```diff
@@ -258,9 +259,9 @@ function init(modules: { typescript: typeof ts_module }) {
                   description: `Disable rule '${problem.getRuleName()}'`,
                   changes: [{
                     fileName: fileName,
-                    textChanges: [{

```

Comment:
I think you can fix this by changing the dependency to typescript in the package.json to typescript@2.3.1-insiders.20170420.  Only this version has the new definition of the Diagnostics type.